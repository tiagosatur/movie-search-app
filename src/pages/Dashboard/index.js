export { default as Dashboard } from './DashBoard'