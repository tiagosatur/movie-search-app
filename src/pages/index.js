export { default as Home } from './Home/Home';
export { default as Movie } from './Movie/Movie';
export { default as Dashboard } from './Dashboard/DashBoard';
export { default as Login } from './Login/Login';