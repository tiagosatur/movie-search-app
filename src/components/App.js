import React from 'react';
import { useSelector } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { PATHS } from '../utils/variables';
import { MainNav, PrivateRoute, Grid } from '.'
import { Home, Movie, Dashboard, Login } from '../pages';
import { GlobalStyle } from '../style/base/global';

const App = () => {
    const {
        user: {
          isLogged,
        },
    } = useSelector(state => state);

    return(
        <div className='app'>
            <GlobalStyle />
            <MainNav />
            <Grid>
                <Switch>
                    <PrivateRoute exact path={PATHS.HOME.PATH} component={Home} />
                    <Route path={PATHS.MOVIE.PATH} component={Movie} />
                    <PrivateRoute isLogged={isLogged} component={Dashboard} /> 
                    <Route path={PATHS.LOGIN.PATH} component={Login} />
                </Switch>
            </Grid>
        </div>
    )
}


export default App;