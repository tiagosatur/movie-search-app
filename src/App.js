import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { PATHS } from './utils/variables';
import { MainNav, PrivateRoute, Grid } from './components'
import { Home, Movie, Dashboard, Login, NotFound } from './pages';
import { GlobalStyle } from './style/base/global';

const App = () => {
    return(
        <div className='app'>
            <GlobalStyle />
            <MainNav />
            <Grid>
                <Switch>
                    <Route exact path={PATHS.HOME.PATH} component={Home} />
                    <PrivateRoute path={PATHS.DASHBOARD.PATH} component={Dashboard} /> 
                    <Route path={PATHS.MOVIE.PATH} component={Movie} />
                    <Route path={PATHS.LOGIN.PATH} component={Login} />
                    <Route component={NotFound} />
                </Switch>
            </Grid>
        </div>
    )
}


export default App;