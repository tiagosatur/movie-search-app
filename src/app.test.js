import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './App';
import { NotFound } from './pages';
import {  MemoryRouter } from "react-router";

describe('App Page tests', () => {
    
    const wrapper =  mount(
        <MemoryRouter initialEntries={[ '/' ]}>
            <App />
        </MemoryRouter>

    )

    it('should call App snapshot correctly', () => {
        const tree = renderer
            .create(wrapper)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('invalid path should redirect to 404/NotFound page', () => {
        const wrapper =  mount(
            <MemoryRouter initialEntries={[ '/ramdom123' ]}>
                <App />
            </MemoryRouter>
    
        )
        expect(wrapper.find(NotFound)).toHaveLength(1);
    });

})


