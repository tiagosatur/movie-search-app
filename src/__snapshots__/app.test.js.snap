// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Page tests should call App snapshot correctly 1`] = `
<div
  className="app"
>
  <nav
    className="main-nav sc-dnqmqq blrMxR"
  >
    <div
      className="sc-gZMcBi cZvPkw"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <h1>
          MovieFlix
        </h1>
      </a>
    </div>
    <ul
      className="sc-iwsKbI tmwtu"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/movies"
          onClick={[Function]}
        >
          Movies
        </a>
      </li>
      <li>
        <a
          href="/dashboard"
          onClick={[Function]}
        >
          Dashboard
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="grid sc-EHOje injBFG"
  >
    <div>
      <h2>
        Home
      </h2>
      <p>
        Wecome to the application main page
      </p>
    </div>
  </div>
</div>
`;
